src/main/java/ch/njol/skript/expressions/ExprMaxFireTicks.java

```diff
@@ -0,0 +1,59 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.util.Timespan;
+import org.bukkit.entity.Entity;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Maximum Time On Fire")
+@Description("Returns the maximum amount of time before an entity begins getting damaged by fire.")
+@Examples({
+	"send target's max time on fire"
+})
+@Since("INSERT VERSION")
+public class ExprMaxFireTicks extends SimplePropertyExpression<Entity, Timespan> {
+
+	static {
+		register(ExprMaxFireTicks.class, Timespan.class, "max[imum] time on fire", "entities");
```

Comment:
I don't think the pattern really reflects what this expression is for. Also, it should allow changing the entity's max fire ticks.

Reply:
What pattern do you think would work better?
Also, I don't think there's a method that allows for it to change

Reply:
I think just staying in line with what Paper calls it works, like `max[imum] fire ticks`. With that being said, the max freeze ticks were done like [this](https://skripthub.net/docs/?id=9492), so I think maybe do it like that.

Reply:
could this possible be merged with the existing burn time class?

Reply:
I think freeze/fire should stay separate, but the freeze/max freeze could potentially be merged?

Reply:
I mean merge the maximum fire tick in with burn time

Reply:
Ah I see. I'd personally vote no, but I don't mind

Reply:
Why? In all fairness it's freezing time which is the odd one out, and these behaviors should be kept together. Majority of the syntax that are separated are older skript versions where newer ones are combined